### ISSUE TITLE
checking for cffi.__version_info__ >= (1,1) in python module cffi... no

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#5457]
```
checking for /usr/bin/python3.6 version... 3.6
checking for /usr/bin/python3.6 platform... linux
checking for /usr/bin/python3.6 script directory... ${prefix}/lib/python3.6/site-packages
checking for /usr/bin/python3.6 extension module directory... ${exec_prefix}/lib64/python3.6/site-packages
checking for cffi.__version_info__ >= (1,1) in python module cffi... no
configure: error: could not find python module cffi, version 1.1+ required
make: *** [Makefile:549: config.status] Error 1
```

ugh, this just started showing up for me on TOSS4 on corona.  I assume the distutils fix did something.  looking into it.

**Comment by chu11** [comment]
i guess `packaging` is not a native python lib/module?  extra?

```
>python3.6
Python 3.6.8 (default, May 31 2023, 10:28:59) 
[GCC 8.5.0 20210514 (Red Hat 8.5.0-18)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import packaging
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'packaging'
>>> import distutils
>>> 
```

**Comment by chu11** [comment]
yup, looks like that was the issue.  will poke admins to install (ahem) "packaging".

**Comment by grondo** [comment]
Sorry, this was my bad. I assumed `packaging` was in the standard library because it is the suggested replacement for `distutils.version` in [PEP 0632](https://peps.python.org/pep-0632/). Apparently they thought it was OK because everyone can just pip anyway so what's the big deal.

We should avoid adding this new requesite package and instead find a way to do the simple version compare within the standard library (or just do something by hand)

**Comment by grondo** [comment]
Looking around it seems like every fix for deprecated `distutils.version` pulls in `packaging.version`, even though it isn't in the standard library. So I guess soon the `packaging` module will be installed everywhere, but since that isn't the case now, the best solution might be to just copy  `distutils` StrictVersion implementation into `am_check_pymod.m4`. :shrug:

**Comment by chu11** [comment]
just a reminder todo to add python3-packaging (or appropriate package name for distro) to the `scripts/install` scripts.

**Comment by grondo** [comment]
I think we're trying to avoid adding a new dependency. However, the best I came up with so far is to try `distutils` and fall back to `packaging` if it can't be imported (basically Python == 3.12). I'm guessing in Python 3.12 everyone will need to have packaging installed so this will be a non-issue? (and I'm not sure there's a distro with Python 3.12 yet, so this is all academic!? :anguished:

**Comment by chu11** [comment]
> I think we're trying to avoid adding a new dependency.

True, we don't need it today.  I guess I was thinking longer out we would.  But we can leave that for another time.

**Comment by grondo** [comment]
Ah, yeah good point

**Comment by grondo** [comment]
fixed by #5459
